### PR TITLE
Fix rate sets to match to the mac80211_hwsim

### PR DESCRIPTION
**Pull request by masap**
The rate sets of mac80211_hwsim is defined at
drivers/net/wireless/mac80211_hwsim.c#hwsim_rates.
The rate sets includes 12 rates.
But wmediumd rateset defined at per.c includes 8 rates.
To fix the mismatch, this patch fixes the wmediumd rateset.

The additional rates are 11b rates, in spite of the wmediumd calculates
SNR with OFDM rates. So we use parameter of closest OFDM rate.

Signed-off-by: Masashi Honma <masashi.honma@gmail.com>

### CODE REVIEW

**Comment by bcopeland** [comment]
This is only true if you use 2.4 ghz frequency, right?

**Comment by masap** [comment]
Ah, right. I only consider 2.4GHz because the test cases (like diamond.sh) uses 2.4GHz channel.

I will fix this patch to support 5GHz also.

**Comment by masap** [comment]
I have fixed and add @patgrosse as a reporter. Thanks !

**Comment by masap** [comment]
Is there any comment ?

**Comment by bcopeland** [comment]
I'm not thrilled about 'set_band()'.  Hmm, I think we could query the phy for the rateset and the interface for the operating frequency...

**Comment by masap** [comment]
@bcopeland 

The frame came from kernel has attribute HWSIM_ATTR_FREQ. So I sent a patch to use it.

File: wmediumd/per.c [hunk 1]
```diff
@@ -25,22 +27,31 @@ struct rate {
...
+	{ .mbps = 60, .mqam = 2, .fec = FEC_RATE_1_2 },
+	{ .mbps = 90, .mqam = 2, .fec = FEC_RATE_3_4 },
+	{ .mbps = 120, .mqam = 4, .fec = FEC_RATE_1_2 },
+	{ .mbps = 180, .mqam = 4, .fec = FEC_RATE_3_4 },
+	{ .mbps = 240, .mqam = 16, .fec = FEC_RATE_1_2 },
+	{ .mbps = 360, .mqam = 16, .fec = FEC_RATE_3_4 },
+	{ .mbps = 480, .mqam = 64, .fec = FEC_RATE_2_3 },
+	{ .mbps = 540, .mqam = 64, .fec = FEC_RATE_3_4 },
 };
+static size_t rate_len = ARRAY_SIZE(rateset);
```
Comment (bcopeland) [inline wmediumd/per.c:52]: not really sure we need this...

File: wmediumd/per.c [hunk 2]
```diff
@@ -25,22 +27,31 @@ struct rate {
...
+/*
+ * rate sets are defined in drivers/net/wireless/mac80211_hwsim.c#hwsim_rates.
+ */
+static struct rate rateset[] = {
+	/*
+	 * XXX:
+	 * For rate = 1, 2, 5.5, 11 Mbps, we will use mqam and fec of closest
+	 * rate. Because these rates are not OFDM rate.
+	 */
+	{ .mbps = 10, .mqam = 2, .fec = FEC_RATE_1_2 },
```
Comment (bcopeland) [inline wmediumd/per.c:39]: should change to .kbps then

**Comment by bcopeland** [comment]
Ok, looks good, thanks.

**Pull request merged**